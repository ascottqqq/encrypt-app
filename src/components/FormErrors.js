import React, {Component} from 'react'

class FormErrors extends Component {

  capitalizeFirstLetter = (string) => {
    return string.charAt(0).toUpperCase() + string.slice(1);
  }

  render() {
    return (
      <div className={this.props.formErrors.hasErrors ? 
          '' : 'text-center'}>
        {this.props.formErrors.hasErrors ? 
          Object.keys(this.props.formErrors).map((fieldName, i) => {
            if(this.props.formErrors[fieldName].length > 0){
              return (
                <p key={i}>{this.capitalizeFirstLetter(fieldName)} 
                {this.props.formErrors[fieldName]}</p>
              )        
            } else {
              return '';
            }
          }) 
        : <p>ChaCha 20</p>}
      </div>
    )
  }
} export default FormErrors
