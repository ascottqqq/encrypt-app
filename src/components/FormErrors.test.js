import React from 'react';
import FormErrors from './FormErrors';
import renderer from 'react-test-renderer';

it('renders without error correctly', () => {
  const tree = renderer.create(
    <FormErrors formErrors={{hasErrors: false, key: '', 
      nonce: '', serverErrors: ''}} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders with key error correctly', () => {
  const tree = renderer.create(
    <FormErrors formErrors={{hasErrors: true, key: ' is invalid', 
      nonce: '', serverErrors: ''}} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders with nonce error correctly', () => {
  const tree = renderer.create(
    <FormErrors formErrors={{hasErrors: true, key: '', 
      nonce: ' is invalid', serverErrors: ''}} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders with server error correctly', () => {
  const tree = renderer.create(
    <FormErrors formErrors={{hasErrors: true, key: '', 
      nonce: '', serverErrors: ' Failed to fetch'}} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders with key and nonce errors correctly', () => {
  const tree = renderer.create(
    <FormErrors formErrors={{hasErrors: true, key: ' is invalid', 
      nonce: ' is invalid', serverErrors: ''}} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders with key and server errors correctly', () => {
  const tree = renderer.create(
    <FormErrors formErrors={{hasErrors: true, key: ' is invalid', 
      nonce: '', serverErrors: ' Failed to fetch'}} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders with nonce and server errors correctly', () => {
  const tree = renderer.create(
    <FormErrors formErrors={{hasErrors: true, key: '', 
      nonce: ' is invalid', serverErrors: ' Failed to fetch'}} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders with key and nonce and server errors correctly', () => {
  const tree = renderer.create(
    <FormErrors formErrors={{hasErrors: true, key: ' is invalid', 
      nonce: ' is invalid', serverErrors: ' Failed to fetch'}} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});