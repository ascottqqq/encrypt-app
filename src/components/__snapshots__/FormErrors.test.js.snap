// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with key and nonce and server errors correctly 1`] = `
<div
  className=""
>
  
  <p>
    Key
     is invalid
  </p>
  <p>
    Nonce
     is invalid
  </p>
  <p>
    ServerErrors
     Failed to fetch
  </p>
</div>
`;

exports[`renders with key and nonce errors correctly 1`] = `
<div
  className=""
>
  
  <p>
    Key
     is invalid
  </p>
  <p>
    Nonce
     is invalid
  </p>
  
</div>
`;

exports[`renders with key and server errors correctly 1`] = `
<div
  className=""
>
  
  <p>
    Key
     is invalid
  </p>
  
  <p>
    ServerErrors
     Failed to fetch
  </p>
</div>
`;

exports[`renders with key error correctly 1`] = `
<div
  className=""
>
  
  <p>
    Key
     is invalid
  </p>
  
  
</div>
`;

exports[`renders with nonce and server errors correctly 1`] = `
<div
  className=""
>
  
  
  <p>
    Nonce
     is invalid
  </p>
  <p>
    ServerErrors
     Failed to fetch
  </p>
</div>
`;

exports[`renders with nonce error correctly 1`] = `
<div
  className=""
>
  
  
  <p>
    Nonce
     is invalid
  </p>
  
</div>
`;

exports[`renders with server error correctly 1`] = `
<div
  className=""
>
  
  
  
  <p>
    ServerErrors
     Failed to fetch
  </p>
</div>
`;

exports[`renders without error correctly 1`] = `
<div
  className="text-center"
>
  <p>
    ChaCha 20
  </p>
</div>
`;
